src/ts/parsers/page.ts

```diff
@@ -0,0 +1,60 @@
+import { PageContent } from "../common/types";
+import {
+  OpenApiBuilder,
+  OpenAPIObject,
+  PathsObject,
+  OperationObject,
+} from "openapi3-ts";
+import { merge as _merge } from "lodash";
+import * as _ from "lodash";
+import * as SwaggerParser from "swagger-parser";
+import debug from "../common/logging";
+
+const debugLog = debug("unmock:parsers:page");
+
+const specBase = {
+  openapi: "3.0.0",
+  info: {
+    version: "1",
+    title: "API title",
+  },
+  paths: {},
+};
+
+const operationBase = (): OperationObject => ({
+  responses: {
+    200: {
+      description: "OK",
+    },
+  },
+});
+
+export const parsePathsFromPage = (pageContent: PageContent): PathsObject => {
+  const pattern = /(GET|POST|DELETE|PUT)\s((?:\/[\w.\/-:{}]*)+)/g;
```

Comment:
A slightly better pattern would be e.g.
`const pattern = /(GET|POST|DELETE|PUT)\s((?:\/[^\/]*?)+)/g;`

Reply:
Care explaining why it's better? Not that I doubt it.

Reply:
Yes of course; two main reasons:
1. Using nested greedy quantifiers is a performance hazard. There's a lot of backtracking that needs to be done to decide on the extent of each quantifiers. Using a lazy quantifier (especially when trying to match small portions in a page content, for example), would be faster. It might even be better to use lazy quantifiers for the external quantifier, but I think it's fine for now.

2. Specifying all the different elements a path item might take seems non-trivial. It holds many assumptions about what we expect to find there, while of course some may decide to use a different charset. Most importantly, using `\/` inside the set would again hinder performance as that's the main delimiter we're matching against. So to allow various structures that maintain a proper path, just match anything that's not a `\/`.

Reply:
Ok thanks, makes sense.

Reply:
Did not work out-of-the-box as it does not capture the whole path but added a note to revisit in future, I'll merge to move forward.

Reply:
Cool, what're the failing cases?

Reply:
Like `GET /v1/pets` captures `GET` and `/` instead of `GET` and `/v1/pets`. Of course it might make sense to first do a big search over the whole document with simple regex and then separate capture, or something.

Reply:
Okay, testing it out and will comment back with a working version :)

Reply:
Please do a PR/push if you do, tbh I don't want to optimize until the extension is published and we have actual users 🙂

Reply:
Fair enough; we can leave it for future PRs. Either way, we'd need to add more options in the future (i.e. `OPTIONS`, `HEAD`, `PATCH`, etc), so this would need a small update :grimacing: